SU2_CFD/src/solvers/CSpeciesFlameletSolver.cpp

```diff
@@ -102,6 +129,20 @@ void CSpeciesFlameletSolver::Preprocessing(CGeometry* geometry, CSolver** solver
                   SU2_MPI::GetComm());
   if ((rank == MASTER_NODE) && (n_not_in_domain_global > 0))
     cout << "Number of points outside manifold domain: " << n_not_in_domain_global << endl;
+
+  /*--- Compute preferential diffusion scalar gradients. ---*/
+  if (config->GetPreferentialDiffusion()) {
+    switch (config->GetKind_Gradient_Method()) {
+      case GREEN_GAUSS:
+        SetAuxVar_Gradient_GG(geometry, config);
```

Comment:
Are communications needed or being handled properly here?

Reply:
They should be. The code is very similar compared to other instances where the auxiliary variable gradients are computed and so far I have not experienced any issues with it.